elastic/kibana
# Implements profile theme selection with KeyPadMenu

Closes #155941

## Summary

This PR changes the implementation of the theme selection component in the user profile screen from EuiButtonGroup to EuiKeyPadMenu. This better matches the original design.

### Previous Implementation
<img width="1090" alt="Screenshot 2023-06-28 at 1 49 22 PM" src="https://github.com/elastic/kibana/assets/103939324/59a184cf-655b-423a-9ba8-16450166fb4c">


### Tests
- `x-pack/plugins/security/public/account_management/user_profile/user_profile.test.tsx`

### Manual Testing
- Start Elasricsearch, and start Kibana with an empty  kibana.yml/kibana.dev.yml
- Navigate to the _Edit profile_ screen via the profile/avatar button in the top right portion of the uI

<img width="269" alt="Screenshot 2023-06-29 at 11 47 42 AM" src="https://github.com/elastic/kibana/assets/103939324/52f5288f-0623-4658-a7a2-2ef85164cadc">

- Verify the theme settings appear as a KeyPadMenu and function as expected

<img width="1079" alt="Screenshot 2023-06-28 at 2 38 44 PM" src="https://github.com/elastic/kibana/assets/103939324/a3745b8b-f908-4cd0-bcf6-2bf878578499">

- Modify the kibana.yml (or kibana.dev.yml) with the line `uiSettings.overrides.theme:darkMode: true`
- Refresh Kibana and verify that the dark theme is rendered and the theme settings are disabled and includes a lock icon tip explaining why the mode setting is locked

<img width="1051" alt="Screenshot 2023-06-30 at 12 24 12 PM" src="https://github.com/elastic/kibana/assets/103939324/5968e1e4-88ae-43a6-8fea-0df7cf2db25b">

- Modify the kibana.yml (or kibana.dev.yml) with the line `uiSettings.overrides.theme:darkMode: false`
- Refresh Kibana and verify that the light theme is rendered and the theme settings are disabled and includes a lock icon tip explaining why the mode setting is locked
